grpc-protocol/src/main/java/com/linecorp/armeria/client/grpc/protocol/UnaryGrpcClient.java

```diff
@@ -164,36 +166,40 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
                        .thenCompose(
                                msg -> {
-                                   final ByteBuf buf = msg.content().byteBuf();
-                                   final HttpData framed;
-                                   try (ArmeriaMessageFramer framer = new ArmeriaMessageFramer(
-                                           ctx.alloc(), Integer.MAX_VALUE, isGrpcWebText)) {
-                                       framed = framer.writePayload(buf);
-                                   }
+                                   try (HttpData content = msg.content()) {
+                                       final ByteBuf buf = content.byteBuf();
+                                       final HttpData framed;
+                                       try (ArmeriaMessageFramer framer = new ArmeriaMessageFramer(
+                                               ctx.alloc(), Integer.MAX_VALUE, isGrpcWebText)) {
+                                           framed = framer.writePayload(buf);
+                                       }
 
-                                   try {
-                                       return unwrap().execute(ctx, HttpRequest.of(req.headers(), framed))
-                                                      .aggregateWithPooledObjects(ctx.eventLoop(), ctx.alloc());
-                                   } catch (Exception e) {
-                                       throw new ArmeriaStatusException(StatusCodes.INTERNAL,
-                                                                        "Error executing request.");
+                                       try {
+                                           return unwrap().execute(ctx, HttpRequest.of(req.headers(), framed))
+                                                          .aggregateWithPooledObjects(ctx.eventLoop(),
+                                                                                      ctx.alloc());
+                                       } catch (Exception e) {
+                                           throw new ArmeriaStatusException(StatusCodes.INTERNAL,
+                                                                            "Error executing request.");
+                                       }
                                    }
                                })
                        .thenCompose(msg -> {
-                           if (!msg.status().equals(HttpStatus.OK) || msg.content().isEmpty()) {
-                               // Nothing to deframe.
-                               return CompletableFuture.completedFuture(msg.toHttpResponse());
-                           }
-
-                           final CompletableFuture<HttpResponse> responseFuture = new CompletableFuture<>();
+                           try (HttpData content = msg.content()) {
+                               if (!msg.status().equals(HttpStatus.OK) || content.isEmpty()) {
```

Comment:
Not your fault but there can be only one `HttpStatus` per each code:
```suggestion
                               if (msg.status() != HttpStatus.OK || content.isEmpty()) {
```

Reply:
Looks much better, thanks!